packages/scheduler/src/forks/SchedulerMock.js

```diff
@@ -195,10 +195,22 @@ function workLoop(hasTimeRemaining, initialTime) {
       const continuationCallback = callback(didUserCallbackTimeout);
       currentTime = getCurrentTime();
       if (typeof continuationCallback === 'function') {
+        // If a continuation is returned, immediately yield to the main thread
```

Comment:
This heuristic is a bit weird because API wise it's only supposed to yield to higher priority but this makes it yield more to same/lower priority native events than before or if you don't use continuation.

It only makes sense if you assume that React only yields a continuation if it has already used up the allotted time because itself has a timer heuristic but other uses of continuations wouldn't necessarily do that.

Reply:
If there were a native way to schedule a continuation, I think how I'd implement it is: inside the reconciler, when exiting the work loop, choose between either native continuation (normal time slice case) or a separate `postTask` call (explicit yield to microtasks). If Scheduler tasks were 1:1 with native tasks that's how I would have done it in this PR, too — we should probably make that change, anyway, so that microtasks aren't blocked by successive tasks and it aligns more closely with `postTask` behavior. But that's a bigger change than I wanted to do within the scope of this PR. So for now I'm treating the continuation API as an internal React detail.

Reply:
I suppose the previous solution (`requestYield`) is closer to that supposed end state, it's just slightly more overhead in the meantime if you assume that React is the only thing that uses this API.